src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,29 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
 
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
+
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                Thread.sleep(200);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
```

Comment:
Using Thread.sleep in an asynchronous operation is not recommended as it blocks the thread unnecessarily. Consider simulating a delay or an asynchronous task with a different approach that does not involve blocking, such as using CompletableFuture's delay methods or scheduling the task with a ScheduledExecutorService.